Gems/ROS2/Code/Source/FactorySimulation/ConveyorBeltComponentKinematic.cpp

```diff
@@ -0,0 +1,306 @@
+                ->Field("BeltWidth", &ConveyorBeltComponentKinematic::m_beltWidth)
+                ->Field("SegmentLength", &ConveyorBeltComponentKinematic::m_segmentLength)
+                ->Field("TextureScale", &ConveyorBeltComponentKinematic::m_textureScale)
+                ->Field("MaterialAsset", &ConveyorBeltComponentKinematic::m_materialAsset);
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<ConveyorBeltComponentKinematic>("Conveyor Belt Component Kinematic", "Conveyor Belt Component")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::EntityId,
+                        &ConveyorBeltComponentKinematic::m_ConveyorEntityId,
+                        "Conveyor Belt Entity",
+                        "Entity of the conveyor belt")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &ConveyorBeltComponentKinematic::m_speed, "Speed", "Speed of the conveyor belt")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &ConveyorBeltComponentKinematic::m_beltWidth, "Width", "Belt width")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ConveyorBeltComponentKinematic::m_segmentLength,
+                        "Segment Length",
+                        "Length of simulated segments. Short segments causes larger number individual bodies to simulate and challenges "
+                        "physics engine.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ConveyorBeltComponentKinematic::m_textureScale,
+                        "Texture scale",
+                        "Scale of the texture on conveyor belt.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ConveyorBeltComponentKinematic::m_materialAsset,
+                        "Material",
+                        "Material asset of the conveyor belt")
+                    ->Attribute(AZ::Edit::Attributes::DefaultAsset, &GetDefaultPhysicsMaterialAssetId)
+                    ->Attribute(AZ_CRC_CE("EditButton"), "")
+                    ->Attribute(AZ_CRC_CE("EditDescription"), "Open in Asset Editor")
+                    ->Attribute(AZ_CRC_CE("DisableEditButtonWhenNoAssetSelected"), true);
+            }
+        }
+    }
+
+    void ConveyorBeltComponentKinematic::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("SplineService"));
+    }
+
+    void ConveyorBeltComponentKinematic::Activate()
+    {
+        m_initialized = false;
```

Comment:
Is the field required? 
Could this be replaced with waiting for the entity to fully activate (OnEntityActivated()), then in the handler connecting to TickBus?

Reply:
Yes.

Reply:
checked, it works.